src/m365/cli/commands/config/config-list.spec.ts

```diff
@@ -0,0 +1,66 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import { telemetry } from '../../../../telemetry';
+import { Cli } from '../../../../cli/Cli';
+import { Logger } from '../../../../cli/Logger';
+import Command from '../../../../Command';
+import { pid } from '../../../../utils/pid';
+import { session } from '../../../../utils/session';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+const command: Command = require('./config-list');
+
+describe(commands.CONFIG_LIST, () => {
+  let log: any[];
+  let logger: Logger;
+  let loggerSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore(Cli.getInstance().config.all);
+  });
+
+  after(() => {
+    sinon.restore();
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.CONFIG_LIST);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('returns a list of all the self set properties', async () => {
+    const config = Cli.getInstance().config;
+    const temp = { ...config.all };
+    config.all = { 'errorOutput': 'stdout' };
```

Comment:
By doing this, you are overriding the config on the local machine. We should avoid this at all costs.
Right now have a stub that doesn't do much. This is because you are using `returns` and this only works for functions. `config.all` is a property instead of a function so we should use another sinon stub.
`sinon.stub(config, 'all').value({ <config> })` should do the trick.

Reply:
Thx, tried resolves, callsfake, returns... Nothing worked.. It always used the environment configuration..
Will try the test like you mentioned. Thank you!